src/components/Flyout/hooks/useOverlayPositionWithBottomMargin.ts

```diff
@@ -14,25 +15,31 @@ type UseOverlayPositionWithBottomMarginProps = {
     overlayRef: MutableRefObject<HTMLElement | null>;
     scrollRef: MutableRefObject<HTMLElement | null>;
     isOpen: boolean;
+    placement?: FlyoutPlacement;
+    offset?: number;
 };
 
 export const useOverlayPositionWithBottomMargin = ({
     triggerRef,
     overlayRef,
     scrollRef,
     isOpen,
+    placement,
+    offset,
 }: UseOverlayPositionWithBottomMarginProps) => {
     const overlayHeight = getTotalOverlayHeight(overlayRef, scrollRef);
 
     const isFlipped = shouldDisplayAbove(triggerRef, overlayHeight, FLYOUT_OVERLAY_OFFSET, INTERCOM_BUTTON_HEIGHT);
+    const verticalPosition = isFlipped ? FlyoutPlacement.Top : FlyoutPlacement.Bottom;
```

Comment:
If `placement = FlyoutPlacement.Top` and `isFlipped = false` it evaluates to Bottom.
The expected behavior would be displaying on top.

![Screenshot 2022-05-23 at 11 29 01](https://user-images.githubusercontent.com/3276065/169789092-4093ed49-9a2c-4de3-861e-449a3e9e237a.png)

Reply:
I fixed it but the flip logic is handled manually as not all the flyout height is within the scroll area. It's calculated for top and bottom but we don't have flip logic done yet for right and left.